File: Leetcode/DataFrom20240701To20241124/T209.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int minSubArrayLen(int target, vector<int>& nums) {
        int CurrentSum = 0;
        int LeftPointer = 0;
        int RightPointer = 0;
        int MinLength = nums.size();
        for(RightPointer = 0;RightPointer < nums.size();RightPointer ++){
            CurrentSum += nums[RightPointer];
            if(CurrentSum >= target){
                while (CurrentSum - nums[LeftPointer] >= target && LeftPointer < RightPointer)
                {
                    CurrentSum -= nums[LeftPointer];
                    LeftPointer ++;
                }
                MinLength = min(MinLength, RightPointer - LeftPointer + 1);
            }
        }
        if(CurrentSum >= target)
            return MinLength;
        else return 0;
    }
};

int main(){

    return 0;
}

File: Leetcode/CodeDataStartFrom20250808/T209.java
public class T209 {
    public int minSubArrayLen(int target, int[] nums) {
        // 注意此题的题目条件。。。是大于等于而不是等于
        
        int n = nums.length, ans = n + 1, cur = 0;
        for (int i = 0, j = 0;j < n;++ j) {
            cur += nums[j];
            if(cur < target) {
                continue;
            }
            while(cur >= target && i <= j) {
                cur -= nums[i ++];
            }
            ans = Math.min(ans, j - i + 2);
        }
        return ans == n + 1 ? 0 : ans;
    }
}
